indexer/packages/postgres/src/query-builders/upsert.ts

```diff
@@ -26,11 +28,16 @@ export default class UpsertQueryBuilder<M extends Model, R = M[]> extends QueryB
     const colBindings = cols.map(() => '??').join(', ');
     const valBindings = cols.map(() => '?').join(', ');
     const setBindings = cols.map(() => '?? = ?').join(', ');
+    const idConditionBindings = idColumns.map(() => '?? = ?').join(' AND ');
```

Comment:
what does ?? = ? do in this case?

Reply:
Positional bindings, see [here](https://knexjs.org/guide/raw.html)